Release 77.0.0 (#1697)
MetaMask/core: packages/transaction-controller/CHANGELOG.md

## [Unreleased]

## [11.0.0]
### Added
- Add optional `getLastBlockVariations` method to `RemoteTransactionSource` type ([#1668](https://github.com/MetaMask/core/pull/1668))

> - Add `updateTransactionGasFees` public method to `TransactionController` ([#1674](https://github.com/MetaMask/core/pull/1674))

Reviewer: ```suggestion
- Add `updateTransactionGasFees` method to `TransactionController` ([#1674](https://github.com/MetaMask/core/pull/1674))
```

> - Add `r`, `s` and `v`  properties to `TransactionMeta` ([#1664](https://github.com/MetaMask/core/pull/1664))

Reviewer: ```suggestion
- Add `r`, `s` and `v`  properties to the transaction metadata ([#1664](https://github.com/MetaMask/core/pull/1664))
```

> - Add `sendFlowHistory` property to transaction meta. ([#1665](https://github.com/MetaMask/core/pull/1665))

Reviewer: ```suggestion
- Add `sendFlowHistory` property the transaction metadata ([#1665](https://github.com/MetaMask/core/pull/1665))
```

Author: should this be like this perhaps?

```suggestion
- Add `sendFlowHistory` property to the transaction metadata. ([#1665](https://github.com/MetaMask/core/pull/1665))
```

Reviewer: Forgot the `to` but I think we're avoiding `.` in this changelog at least.

> - Add `updateTransactionSendFlowHistory` function to the public API of the controller. ([#1665](https://github.com/MetaMask/core/pull/1665))

Reviewer: ```suggestion
- Add `updateTransactionSendFlowHistory` method to `TransactionController` ([#1665](https://github.com/MetaMask/core/pull/1665))
```

> - Persist `originalGasEstimate` in transaction metadata ([#1656](https://github.com/MetaMask/core/pull/1656))

Reviewer: ```suggestion
- Add `originalGasEstimate` property to the transaction metadata ([#1656](https://github.com/MetaMask/core/pull/1656))
```

### Changed

> - **BREAKING**: Remove `apiKey` property from `RemoteTransactionSourceRequest` type ([#1668](https://github.com/MetaMask/core/pull/1668))

Reviewer: ```suggestion
- **BREAKING**: Remove `apiKey` property from `RemoteTransactionSourceRequest` type ([#1668](https://github.com/MetaMask/core/pull/1668))
  - Not currently used in either client.
```

Author: AFAIK we don't reference usage in _clients_(referring to the extension or mobile for example), as a client could be any lib dependant on this package

- **BREAKING**: Remove unused `FetchAllOptions` type from `TransactionController` ([#1668](https://github.com/MetaMask/core/pull/1668))

> - **BREAKING**: Remove `incomingTransactions.apiKey` property from `TransactionController` constructor arguments ([#1668](https://github.com/MetaMask/core/pull/1668))

Reviewer: ```suggestion
- **BREAKING**: Remove `incomingTransactions.apiKey` constructor option ([#1668](https://github.com/MetaMask/core/pull/1668))
```

> - **BREAKING**: Rename the `transaction` object to `txParams` in the persisted `TransactionMeta` ([#1651](https://github.com/MetaMask/core/pull/1651))

Reviewer: ```suggestion
- **BREAKING**: Rename the `transaction` object to `txParams` in the transaction metadata ([#1651](https://github.com/MetaMask/core/pull/1651))
```

> - **BREAKING**: Added `disableHistory` constructor option ([#1657](https://github.com/MetaMask/core/pull/1657))

Reviewer: ```suggestion
- **BREAKING**: Add `disableHistory` constructor option ([#1657](https://github.com/MetaMask/core/pull/1657))
  - Defaults to `false` but will increase state size considerably unless disabled.
```

> - **BREAKING**: Added `disableSendFlowHistory` constructor option ([#1665](https://github.com/MetaMask/core/pull/1665))

Reviewer: ```suggestion
- **BREAKING**: Add `disableSendFlowHistory` constructor option ([#1665](https://github.com/MetaMask/core/pull/1665))
  - Defaults to `false` but will increase state size considerably unless disabled.
```

> - **BREAKING**: Rename the `transactionHash` property to `hash` in the persisted `TransactionMeta`.

Reviewer: ```suggestion
- **BREAKING**: Rename the `transactionHash` property to `hash` in the transaction metadata
```

- Add the `incomingTransactions.queryEntireHistory` constructor option.

> - Only use the `incomingTransactions.apiKey` when querying the standard Etherscan domain. ([#1652](https://github.com/MetaMask/core/pull/1652))

Reviewer: Can be removed as this is superseded by a later commit where the API key was removed.

### Fixed

> - Fix the sorting of incoming and updated transactions ([#1652](https://github.com/MetaMask/core/pull/1652))

Reviewer: I forgot to include it in [this PR](https://github.com/MetaMask/core/pull/1668) but worth adding here:

- Prevent rate limit errors when `incomingTransactions.includeTokenTransfers` is `true` by by alternating Etherscan request types on each update

- Add `originalGasEstimate` property to the transaction metadata ([#1656](https://github.com/MetaMask/core/pull/1656))

### Changed
- **BREAKING**: Remove `apiKey` property from `RemoteTransactionSourceRequest` type ([#1668](https://github.com/MetaMask/core/pull/1668))
- **BREAKING**: Remove unused `FetchAllOptions` type from `TransactionController` ([#1668](https://github.com/MetaMask/core/pull/1668))
- **BREAKING**: Remove `incomingTransactions.apiKey` constructor option ([#1668](https://github.com/MetaMask/core/pull/1668))
- **BREAKING**: Rename the `transaction` object to `txParams` in the transaction metadata ([#1651](https://github.com/MetaMask/core/pull/1651))
- **BREAKING**: Add `disableHistory` constructor option ([#1657](https://github.com/MetaMask/core/pull/1657))
  - Defaults to `false` but will increase state size considerably unless disabled
- **BREAKING**: Add `disableSendFlowHistory` constructor option ([#1665](https://github.com/MetaMask/core/pull/1665))
  - Defaults to `false` but will increase state size considerably unless disabled
- **BREAKING**: Rename the `transactionHash` property to `hash` in the transaction metadata

> - Add `incomingTransactions.queryEntireHistory` constructor option

Reviewer 2: It's a really small one, move this line to the `Added` section. [PR](https://github.com/MetaMask/core/pull/1652)
